### PR TITLE
Update some docs for v0.3.0

### PR DESCRIPTION
**Pull request by twsouthwick**

### CODE REVIEW

**Comment by Code-DJ** [comment]
@twsouthwick in the readme, is AddCompiledPages always needed? 

```
.AddDynamicPages() // Remove if you don't have dynamic pages
.AddCompiledPages();
```

**Comment by twsouthwick** [comment]
Right now it's either/or - not sure they'd play nice together.

I'm considering a source generator with an interceptor that will allow just doing

```
.AddPages()
```

and have it decide at compile time what to do. Not sure though....

**Comment by twsouthwick** [comment]
@Code-DJ here's an issue to track that https://github.com/CoreWebForms/CoreWebForms/issues/250

File: docs/designs/Compilation.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Page compilation
+
+The compilation system for ASP.NET Framework was MSBuild based and not something that we needed to bring-forward as-is as it was mostly an implementation detail. Instead, we are using Roslyn and creating compilations based on that.
+
+## IWebFormsCompiler
+
+The `IWebFormsCompiler` interface is used to access the heavy lifting of taking a collection of input files and compiling that aspx/ascx/etc into their appropriate types. It takes an `ICompilationStrategy` that is used to configure what the compilation should look like. The main implementations of this are for in-memory (i.e. runtime) compilation compared to static compilation (i.e. aspnet_compiler).
```
Comment (birojnayak) [inline docs/designs/Compilation.md:7]: You probably wants to have another section explainaing  ICompilationStrategy. 

Also see if you want to follow [this style](https://github.com/CoreWCF/CoreWCF/blob/30a6ce9c86332bb11f20eccd2ba84404f04b7165/Documentation/DesignDocs/transport_message_cred_using_cert.md#iservicedispatcher).(This is lot of work, so optional :))

File: docs/designs/Compilation.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+## PageCompilationOptions
+
+This type is used to configure various aspects of the general compilation system. This includes:
+
+- *Parsers* - The implementations of various `DependencyParser` for a given extension. This type has been augmented from the ASP.NET Framework version to include the ability to generate a code parser and then a code generator from that as a starting point.
```
Comment (birojnayak) [inline docs/designs/Compilation.md:13]: See if you can add about determining compilation dependency and how we sort them before invoking parsers/generators.

**Review by birojnayak — COMMENTED** [review]
in general looks ok, but providing little more info would help other users I think... see what you think.

File: docs/designs/Compilation.md (hunk 1 above)
Comment (twsouthwick) [reply docs/designs/Compilation.md:7]: great style, but feels like too much work right now. The goal of these docs at the moment is to be able to point people to where they should look for more details.

**Comment by twsouthwick** [comment]
@birojnayak added a bit of more information - feel free to add more if you think it needs it.

**Review by birojnayak — APPROVED** [review]
LGTM

**Pull request merged**